src/request/TopLevelApi.js

```diff
@@ -60,8 +60,27 @@ class TopLevelApi extends RequestParser { // eslint-disable-line no-unused-vars
          *
          * @deprecated Only for database migration
          */
-        if ((BrowserDetection.isIOS() || BrowserDetection.isSafari()) && TopLevelApi._hasMigrateFlag()) {
-            await KeyStore.instance.migrateAccountsToKeys();
+        if ((BrowserDetection.isIOS() || BrowserDetection.isSafari())) {
+            if (TopLevelApi._hasMigrateFlag()) {
+                await KeyStore.instance.migrateAccountsToKeys();
+            }
+            if (TopLevelApi._hasRemoveKey()) {
```

Comment:
What is this for?

Reply:
There is a case when an imported secret is of type unknown, either entropy or private key. In that case both get encrypted and stored to the Database. When the account finding in accounts manager finds only one of the two accounts has use, it will try and delete the other.

Reply:
Ha! Tricky. If it is not already explained in a comment somewhere, that would be nice.

Reply:
I very much agree.